scripts/ci/validatecode.ps1

```diff
@@ -163,178 +196,228 @@ $HardcodedPathExceptions = @{
+        $containsIssue = $false
 
-    $MatchesPlayspaces = Select-String MixedRealityPlayspace $FileName -AllMatches
-    $NumPlayspaces = $MatchesPlayspaces.Matches.Count
+        $MatchesPlayspaces = Select-String MixedRealityPlayspace $FileName -AllMatches
+        $NumPlayspaces = $MatchesPlayspaces.Matches.Count
 
-    if ($NumPlayspaces -gt 1){
-        Write-Host "There are multiple MixedRealityPlayspace objects in $FileName, delete the extra playspaces from the unity scene."
-        $containsIssue = $true
-    }
+        if ($NumPlayspaces -gt 1){
+            Write-Warning "There are multiple MixedRealityPlayspace objects in $FileName, delete the extra playspaces from the unity scene."
+            $containsIssue = $true
+        }
 
-    if (CheckHardcodedPath $FileName) {
-        $containsIssue = $true
-    }
+        if (CheckHardcodedPath $FileName) {
+            $containsIssue = $true
+        }
 
-    if (CheckForMetaFile $FileName) {
-        $containsIssue = $true
-    }
+        if (CheckForMetaFile $FileName) {
+            $containsIssue = $true
+        }
 
-    return $containsIssue
+        $containsIssue
+    }
 }
 
-Write-Output "Checking $Directory for common code issues"
-
-$codeFiles = Get-ChildItem $Directory *.cs -Recurse | Select-Object FullName
-$containsIssue = $false
-foreach ($codeFile in $codeFiles) {
-    if (CheckScript $codeFile.FullName) {
-        $containsIssue = $true
+# If the file containing the list of changes was provided and actually exists,
+# this validation should scope to only those changed files.
+if ($ChangesFile -and (Test-Path $ChangesFile -PathType leaf)) {
+    # TODO(https://github.com/microsoft/MixedRealityToolkit-Unity/issues/7022)
+    # There may be ways to configure common modules so that paths like this aren't required
+    Import-Module -Force (Resolve-Path("$RepoRoot\scripts\ci\common.psm1"))
+
+    Write-Host "Checking only changed files for code issues: $ChangesFile"
+    $changedFiles = GetChangedFiles -Filename $ChangesFile -RepoRoot $RepoRoot
+    ForEach ($changedFile in $changedFiles) {
+        Write-Host "Checking file: $changedFile"
+        if (((IsCSharpFile -Filename $changedFile) -and (CheckScript $changedFile)) -or
+            ((IsAssetFile -Filename $changedFile) -and (CheckAsset $changedFile)) -or
+            ((IsUnityScene -Filename $changedFile) -and (CheckUnityScene $changedFile)) -or
+            ((IsMetaFile -Filename $changedFile) -and (CheckForActualFile $changedFile))) {
+            $containsIssue = $true;
+        }
     }
 }
+else {
+    Write-Host "Checking $Directory for common code issues"
 
-$codeFiles = Get-ChildItem $Directory *.asset -Recurse | Select-Object FullName
-foreach ($codeFile in $codeFiles) {
-    if (CheckAsset $codeFile.FullName) {
-        $containsIssue = $true
+    $codeFiles = Get-ChildItem $Directory *.cs -Recurse | Select-Object FullName
```

Comment:
Perhaps this entire "else" could be removed as it seems to do the same thing as the code above. The script could set $changedFiles to either the paths from the text file or from Get-ChildItem and then run the ForEach (line 380 above) in either case (just on a different set of entries in the changedFiles array).

Reply:
Perhaps the file patterns make else more memory efficient but the check that matches file type to an extension is repeated in two places (times 2 scripts), so if an update is needed at any point it is easier to oversee this.
Anyway, it's just a suggestion. :)

Reply:
Ack - I'm going to add this to my list of things that I'll tackle coming up, but I'll opt to keep it out of this change. I feel like I've gotten this change what is a working state and there's a decent amount of testing that has to be reset if I change this stuff up.

Your suggestion is a good one - next time I do another pass on this I can include the change.